internal/telemetry/event.go

```diff
@@ -0,0 +1,66 @@
+package telemetry
+
+import (
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Event is a telemetry event
+type Event struct {
+	id          string
+	eventType   EventType
+	userID      string
+	time        time.Time
+	executionID string
+	command     string
+	data        map[DataKey]interface{}
+}
+
+// NewCommandStartEvent creates a new event for a command's start
+func NewCommandStartEvent(command string) Event {
+	data := make(map[DataKey]interface{})
+	return NewEvent(EventTypeCommandStart, command, data)
+}
+
+// NewCommandCompleteEvent creates a new event for a command's start
+func NewCommandCompleteEvent(command string) Event {
+	data := make(map[DataKey]interface{})
+	return NewEvent(EventTypeCommandStart, command, data)
+}
+
+// NewCommandErrorEvent creates a new event for a command that has errored
+func NewCommandErrorEvent(command string, err error) Event {
```

Comment:
I think we can get rid of these three constructor functions, and simply just use `NewEvent` here.

For the `data` arg, we could leverage a variadic type so that the `ErrorEvent` can still pass in its error for now